File: scripts/clean.mjs
import { readdir, rm } from "node:fs/promises";
import { resolve } from "node:path";

const root = resolve(process.cwd());
const publicDir = resolve(root, "public");
const targets = [resolve(root, "dist"), resolve(root, "generated"), resolve(root, ".cache"), resolve(root, ".vite")];
const removeTarget = async (target) => rm(target, { recursive: true, force: true });
const removeGeneratedPublicPdfs = async () => {
  let entries = [];
  try { entries = await readdir(publicDir, { withFileTypes: true }); } catch { return; }
  await Promise.all(entries.filter((entry) => entry.isDirectory()).map((entry) => removeTarget(resolve(publicDir, entry.name, "articles"))));
};
await Promise.all(targets.map(removeTarget));
await removeGeneratedPublicPdfs();
console.log("clean_ok: removed dist, generated, caches, and generated public PDFs");

File: scripts/clean.py
"""
Build cleanup utilities.

Removes generated output, build directories, caches, generated public PDFs,
and generated public media assets.
"""

from __future__ import annotations

import logging
import shutil
from pathlib import Path

from .config import DIST_DIR, GENERATED_DIR, PUBLIC_DIR, ROOT_DIR

logger = logging.getLogger(__name__)


class Clean:
    targets = (
        DIST_DIR,
        GENERATED_DIR,
        ROOT_DIR / ".cache",
        ROOT_DIR / ".vite",
    )

    def run(self) -> None:
        for target in self.targets:
            self.remove(target)

        self.remove_generated_public_files()
        logger.info("Removed generated output, caches, public PDFs and media assets.")

    @staticmethod
    def remove(path: Path) -> None:
        shutil.rmtree(path, ignore_errors=True)

    def remove_generated_public_files(self) -> None:
        if not PUBLIC_DIR.exists():
            return

        for pdf in PUBLIC_DIR.rglob("*.pdf"):
            pdf.unlink()

        self.remove(PUBLIC_DIR / "media")


def run() -> None:
    Clean().run()
